blogs/treewidth.md

```diff
@@ -0,0 +1,385 @@
+
+For the second question, we have the following two reasons:
+
+1. Width of the tree decomposition is an upper bound of the space complexity of the corresponding contraction.
+2. Total dimension of the indices in the same tree bag is the time complexity of the corresponding step of the contraction.
+
+For example, in the first step of the construction above, the index $a$ is eliminated, and the corresponding operations is
+$$A'_{bcde} = \sum_{a} A_{abc} B_{cde}$$
+where the space complexity is bounded by width and the time complexity is given by the dimension of the indices in bag.
+
+Thus, finding the tree decomposition with minimal width is equivalent to finding the optimal contraction order of the tensor network, with a bounded space complexity and minimal time complexity.
+
+## The Bouchitté–Todinca Algorithm for Tree Width
+
+In the previous sections, we introduced the connection between tree decomposition and contraction order, and the importance of finding the tree decomposition with minimal width. 
+Thus, next question is: **How to find the tree decomposition with minimal width?**
+
+In previous few decades, methods have been proposed to find the tree decomposition with minimal width, including exact algorithms and heuristic algorithms.
+In this section, we will introduce the Bouchitté–Todinca (BT) algorithm[^Bouchitté] for finding the tree decomposition with minimal width.
+
+Generally speaking, the BT algorithm is a dynamic programming algorithm. 
+Given a set of possible tree bags, the algorithm can search a set of tree bags forming a tree decomposition with minimal width.
+It has been proved that for a decomposition with minimal width, all bags are **potential maximal cliques** (pmc).
+Thus the BT algorithm contains two phases, it first search for all pmcs, and then search the minimal width decomposition with the pmcs.
+
+### Minimum Separator and Potential Maximal Cliques
+
+To introduce the algorithm, we first need to introduce the concept of **minimum separator**[^Berry] and **potential maximal cliques**[^BouchittéListing], which are the intersection of two bags and possible tree bags, respectively.
+
+**Definition 2**. A subset $S \subseteq V$ is an $a, b$-separator for two nonadjacent vertices $a, b \in V$ if the removal of $S$ from the graph separates $a$ and $b$ in different connected components. $S$ is a minimal $a, b$-separator if no proper subset of $S$ separates $a$ and $b$.
+
+For all pairs of nonadjacent vertices $a, b \in V$, the set of all minimal $a, b$-separators are called the minimal separators of the graph, represented by $\Delta(G)$.
+In practice, we can examine a set of vertices $S$ by checking properties of the connected components of the graph after removing $S$, represented as $\mathcal{C}(G / S)$:
+* If there are two or more connected components $C \in \mathcal{C}(G / S)$ whose neighbor is exactly the set $S$, then $S$ is a separator, and the components are called **full components**.
```

Comment:
I don' think it has been explained what it means $S$ to be a neighbour of one of the connected components from $\mathcal{C}(G / S)$